### PR TITLE
Process incoming packets on different interfaces in a round robin fashion.

### PR DESCRIPTION
**Pull request by liamjm**
The current approach selects the first endpoint  / interface (the main FIDO one) first all the time. This introduces a delay in processing the Vendor HID interface. This can lead to timeouts in processing the packets as each packet must be received within 100ms of the last one. #502 addresses this by increasing this timeout, but this is only a hack.

This PR fixes this problem by always processing the other interface than the previous packet. This is implemented by tracking the last endpoint that returned data, and trying the other one first next time (new next_endpoint_index variable).
This PR is implemented as a change to an existing patch (patches/tock/12-pending-deadlock.patch) as this is this patch has overlapping functionality.

### CODE REVIEW

File: patches/tock/12-pending-deadlock.patch [hunk 1]
```diff
@@ -50,25 +52,38 @@ index 16b80cb10..c71ed639a 100644
...
-     }
+@@ -352,6 +357,13 @@ impl<'a, 'b, C: hil::usb::UsbController<'a>> ClientCtapHID<'a, 'b, C> {
+                 self.cancel_in_transaction(endpoint);
  
+                 self.client.map(|client| client.packet_received(&buf, endpoint, app));
++                // Update next packet to send.
++                for (i, ep) in self.endpoints.iter().enumerate() {
++                  if ep.endpoint == endpoint {
++                    self.next_endpoint_index.set((i + 1) % NUM_ENDPOINTS);
++                    break
```
Comment (ia0) [inline patches/tock/12-pending-deadlock.patch]: nit: semi-colon after `break` (and other expressions used as statement that return `!`, i.e. any type, but don't rely on it, i.e. unit `()` would suffice).

**Review by ia0 — DISMISSED** [review]

File: patches/tock/12-pending-deadlock.patch [hunk 2]
```diff
@@ -50,25 +52,38 @@ index 16b80cb10..c71ed639a 100644
 +            // The previous packet has not yet been received, reject the new one.
 +        } else {
 +            self.pending_out.set(true);
-+            // Process the first endpoint that has a delayed packet.
-+            // TODO(liamjm): Determine if a round-robin approach reduces latency
-+            // in the worst case.
-+            for (_, s) in self.endpoints.iter().enumerate() {
++            // Process the next endpoint that has a delayed packet.
++            for i in self.next_endpoint_index.get()..self.next_endpoint_index.get()+NUM_ENDPOINTS {
```
Comment (kaczmarczyck) [inline patches/tock/12-pending-deadlock.patch]: nit: Some formatting is off, like the spaces around + here.

**Review by kaczmarczyck — DISMISSED** [review]

File: patches/tock/12-pending-deadlock.patch (hunk 1 above)
Comment (liamjm) [reply patches/tock/12-pending-deadlock.patch]: Done.

**Review by ia0 — DISMISSED** [review]

File: patches/tock/12-pending-deadlock.patch [hunk 3]
```diff
@@ -1,9 +1,9 @@
 diff --git a/capsules/src/usb/usbc_ctap_hid.rs b/capsules/src/usb/usbc_ctap_hid.rs
-index 16b80cb10..2b19519e4 100644
+index 16b80cb10..949388b70 100644
 --- a/capsules/src/usb/usbc_ctap_hid.rs
 +++ b/capsules/src/usb/usbc_ctap_hid.rs
 @@ -120,7 +120,6 @@ struct EndpointState {
-
+
```
Comment (kaczmarczyck) [inline patches/tock/12-pending-deadlock.patch]: This commit introduced a bunch of spaces for some reason?

**Review by kaczmarczyck — DISMISSED** [review]

File: patches/tock/12-pending-deadlock.patch (hunk 3 above)
Comment (liamjm) [reply patches/tock/12-pending-deadlock.patch]: :shrug: 
Perhaps these came from an IDE.

Removed.

**Review by kaczmarczyck — APPROVED** [review]

**Pull request merged**